tests/unittests/test_datasource/test_azure.py

```diff
@@ -526,7 +526,7 @@ def _get_mockds(self):
         ])
         return dsaz
 
-    def _get_ds(self, data, agent_command=None, distro=None,
+    def _get_ds(self, data, agent_command=None, distro='ubuntu',
```

Comment:
there are a few tests that actually do not set distros. I changed the ones that explicitly set distros=None to distros=mock.Mock(). For the one that do not set distros it was easier to have them default to something (since in reality distro is never None anyway, better have a more meaningful default)